doozerlib/cli/__main__.py

```diff
@@ -379,34 +380,33 @@ def images_update_dockerfile(runtime, stream, version, release, repo_type, messa
     metas = runtime.image_metas()
     lstate['total'] = len(metas)
 
-    for image in metas:
+    def dgr_update(image_meta):
         try:
-            dgr = image.distgit_repo()
+            dgr = image_meta.distgit_repo()
             (real_version, real_release) = dgr.update_distgit_dir(version, release)
             dgr.commit(message)
             dgr.tag(real_version, real_release)
-            state.record_image_success(lstate, image)
+            state.record_image_success(lstate, image_meta)
+            if push:
+                (meta, success) = dgr.push()
+                if success is not True:
+                    state.record_image_fail(lstate, meta, success)
         except Exception as ex:
             msg = None
             try:
                 msg = ex.strerror
             except:
                 msg = str(ex)
 
-            state.record_image_fail(lstate, image, msg, runtime.logger)
-
-    try:
-        if push:
-            res = runtime.push_distgits()
-
-            for name, r in res:
-                if r is not True:
-                    state.record_image_fail(lstate, name, r)
-    except Exception as ex:
-        lstate['status'] = state.STATE_FAIL
-        raise DoozerFatalError(getattr(ex, 'message', repr(ex)))
+            state.record_image_fail(lstate, image_meta, msg, runtime.logger)
+            return False
+        return True
 
-    state.record_image_finish(lstate)
```

Comment:
i'm not seeing where this got moved to. i'm not certain what it does but it seems significant.

Reply:
Nice. Next push.